engines/supernova/supernova.cpp

```diff
@@ -407,6 +413,10 @@ void SupernovaEngine::setTextSpeed() {
 	_gm->animationOn();
 
 	_improved = hasImprovedMode;
+
+	keymapper->getKeymap("menu")->setEnabled(false);
+	keymapper->getKeymap("supernova-default")->setEnabled(true);
+	keymapper->getKeymap("improved-mode")->setEnabled(true);
```

Comment:
I am wondering, would it make sense to only enable this keymap if `_improved` is true rather than check for `_improved` in `GameManager::processInput`?

Reply:
My suggestion was to replace `keymapper->getKeymap("improved-mode")->setEnabled(true);` with `keymapper->getKeymap("improved-mode")->setEnabled(_improved);` (or `_vm->_improved`). But that was a minor suggestion which I think might be a little bit cleaner, and I think your code is fine as it is. So unless another reviewer thinks it would be better to change it, I think you can leave it as it is currently.